### PR TITLE
Support Slot-level Targeting

### PR DESCRIPTION
**Pull request by ericsun977**
This PR accomplishes:

1. Add a new API `setSlotTargeting` will set targeting information to the Ad slot
2. Set a new watcher for getting Series
3. Get the position from the directive
4. Refresh the Ad when we get all the info.

### CODE REVIEW

File: angular-dfp.js [hunk 1]
```diff
@@ -426,6 +444,27 @@ angular.module('ngDfp', [])
...
             }
 
+            scope.$watch('target', function(target) {
+              if(angular.isUndefined(target) || target.length < 1) {
+                return;
+              }
+
+              var targetingInfo = JSON.parse(target);
+
+              // Set the position of the ad
```
Comment (Tidwell) [inline angular-dfp.js]: If I'm reading this right, could pos and env both be passed params from the client app instead?  Do we definitely want those targeting params set in every project that uses this lib?

File: angular-dfp.js [hunk 2]
```diff
@@ -426,6 +441,21 @@ angular.module('ngDfp', [])
               });
             }
 
+            scope.$watch('target', function(target) {
+              if(angular.isUndefined(target) || target.length < 1) {
+                return;
+              }
+
+              var targetingInfo = JSON.parse(target);
```
Comment (Tidwell) [inline angular-dfp.js]: Does this have to be json stringify/parsed, is it unable to pass an object through?  Not a big deal, just seems a little odd.

Comment (ericsun977) [inline angular-dfp.js]: Yeah I know, it looks odd when we are so used to use Angular component...

Comment (Tidwell) [inline angular-dfp.js]: okies 👍

**Pull request merged**